### PR TITLE
#167190448 create tags and categories

### PR DESCRIPTION
**Pull request by Lundii**
#### What does this PR do?
- it adds tags and/or categories to and article when created
- it creates tags and/or categories to the database
- it gets all tags
- it gets all categories

#### Description of the task to be completed.
- created the endpoint `GET /api/v1/articles/tags/get`
- created the endpoint `GET /api/v1/articles/categories/get`
- created the endpoint `POST /api/v1/articles/tags/create`
- created the endpoint `POST /api/v1/articles/categories/create`
- used the create article endpoint to add tags and/or categories to an article

#### How can this be manually tested?
- fetch or pull this branch
- run npm install to get all packages
- visit `POST /api/v1/articles` ,create an article an include a tagList and/or categoryList field to create either or both of them
- visit `GET /api/v1/articles/tags/get` to get all tags in the database
- visit `GET /api/v1/articles/categories/get` to get all categories in the database
- visit `POST /api/v1/articles/tags/create` to create a new tag in the database
```
{
  tagName: 'something',
 description: 'can be omitted'
}
```
- visit `POST /api/v1/articles/categories/create` to create a new category in the database

```
{
  categoryName: 'something',
 description: 'can be omitted'
}
```

#### What are the relevant Pivotal Tracker stories?
[#167190448](https://www.pivotaltracker.com/story/show/167190448)

### CODE REVIEW

File: server/tests/article.test.js [hunk 1]
```diff
@@ -270,7 +284,21 @@ describe('Article test', () => {
       });
   });
 
-  it('no tagList: should throw an error if tagList is not provided', (done) => {
+  // it('should return 400 if taglist does not follow specified format', (done) => {
```
Comment (chuxmykel) [inline server/tests/article.test.js]: This should be deleted if it's not necessary.

**Review by chuxmykel — APPROVED** [review]
Looks good to me

File: server/tests/article.test.js (hunk 1 above)
Comment (chiboycalix) [reply server/tests/article.test.js]: Yea.

**Review by topseySuave — APPROVED** [review]

File: server/tests/article.test.js (hunk 1 above)
Comment (Lundii) [reply server/tests/article.test.js]: Thanks for the observation will do that.

**Review by obayomi96 — APPROVED** [review]
LGTM

**Review by chiboycalix — APPROVED** [review]
LGTM

**Review by chuxmykel — APPROVED** [review]
LGTM

**Pull request merged**